src/server/transaction.h

```diff
@@ -166,9 +166,11 @@ class Transaction {
   // Can be used only for single key invocations, because it writes a into shared variable.
   template <typename F> auto ScheduleSingleHopT(F&& f) -> decltype(f(this, nullptr));
 
-  // Called by EngineShard when performing Execute over the tx queue.
+  // Called by engine shard to execute a transaction hop.
+  // txq_ooo is set to true if the transaction is running out of order
+  // not as the tx queue head.
   // Returns true if transaction should be kept in the queue.
-  bool RunInShard(EngineShard* shard);
+  bool RunInShard(EngineShard* shard, bool txq_ooo);
```

Comment:
Maybe I should invert the parameter and call it txq_head? (which would implicitly include continuation_trans_ and awakened ones)

Reply:
I actually did similar thing with https://github.com/dragonflydb/dragonfly/pull/1376
I think it's clearer this way. 
I know it's not needed for the fix but consider adding `RUN_AS_OOO` to local_mask for debugging purposes.